### ISSUE TITLE
Unable to use tree view with sbt BSP

### ISSUE THREAD

**Issue by ckipp01** [scalameta/metals#2204]
**Describe the bug**

When using sbt BSP, if you try to dive down into the tree view panel, compilation goes nuts. I'm not really sure what's happening yet and will need to dive into this, but wanted to submit this just in case someone else comes across it.

**To Reproduce**

1. Open a project
2. Switch to sbt BSP
3. Make sure project is compiled
4. Open the side panel and try to drill down into the package
5. Constant compilations happens

**Expected behavior**

Not constant compilation

**Screenshots**

![2020-11-11 11 07 07](https://user-images.githubusercontent.com/13974112/98799057-e8def500-240e-11eb-8799-85bfc53e1621.gif)

**Installation:**

- Operating system: macOS
- Editor: Tried on both VS Code and coc-metals
- Metals version: v0.9.5

**Search terms**

Tree view sbt BSP

**Comment by tgodzik** [comment]
Thanks for finding this! I can confirm it happens also in my case. The problem is probably something around https://github.com/scalameta/metals/blob/48b20201f3357738ce472c40dcae47b0cccc56fb/metals/src/main/scala/scala/meta/internal/tvp/MetalsTreeViewProvider.scala#L69-L77

But it seems super weird.

**Comment by ckipp01** [comment]
So I'm having a way harder time than expected tracking this one down.

Am I correct in understanding that the only reason the `doCompile(id)` in the snippet up above exists is to populate the tree if using Bloop and you have not yet triggered a compilation, but open the tree view? I'm still trying to figure out what exactly is causing the cyclical call to compile, but one of the first things I tried was just commenting the `doCompile(id)` out, and everything worked as expected with both Bloop and sbt _except_ in the situation where I opened up the `build.sbt` only, and then opened the treeview while using Bloop to find it empty until I navigated to a build target.
